ckan/logic/action/update.py

```diff
@@ -387,47 +409,56 @@ def package_update(
             and plugin_data
         )
 
-    pkg = model_save.package_dict_save(
+    nested = model.repo.session.begin_nested()
```

Comment:
Can you explain why this is needed?

Reply:
This lets us commit or rollback just the changes to this package without affecting the outer transaction in case `defer_commit=True` is passed in the context